source/extensions/filters/http/aws_lambda/aws_lambda_filter.cc

```diff
@@ -349,6 +360,9 @@ Http::FilterDataStatus AWSLambdaFilter::decodeData(Buffer::Instance &data,
   }
 
   if (end_stream) {
+    if (isRequestTransformationNeeded() && has_body_) {
```

Comment:
why do we check for body here? 
transformations in general do not require a body?

Reply:
please add unit test to prevent regressions